test/eofparse/eofparse.cpp

```diff
@@ -0,0 +1,85 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <evmc/evmc.hpp>
+#include <evmone/eof.hpp>
+#include <iostream>
+#include <string>
+
+namespace
+{
+inline constexpr bool isalnum(char ch) noexcept
+{
+    return (ch >= '0' && ch <= '9') || (ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z');
+}
+
+template <typename BaseIterator>
+struct skip_nonalnum_iterator : evmc::filter_iterator<BaseIterator, isalnum>
+{
+    using evmc::filter_iterator<BaseIterator, isalnum>::filter_iterator;
+};
+
+template <typename BaseIterator>
+skip_nonalnum_iterator(BaseIterator, BaseIterator) -> skip_nonalnum_iterator<BaseIterator>;
+
+template <typename InputIterator>
+std::optional<evmc::bytes> from_hex2(InputIterator begin, InputIterator end) noexcept
+{
+    evmc::bytes bs;
+    if (!from_hex(skip_nonalnum_iterator{begin, end}, skip_nonalnum_iterator{end, end},
+            std::back_inserter(bs)))
+        return {};
+    return bs;
+}
+
+[[maybe_unused]] auto from_hex2(std::string_view s) noexcept
+{
+    return from_hex2(s.begin(), s.end());
+}
+
+}  // namespace
+
+int main()
```

Comment:
I think this helper tool and the fuzzer could be split into a separate PR and jut be merged that way.

Reply:
See #568